Automattic/wp-calypso
# DIFM Signup Flow, Design picker V2 Integration

### Changes proposed in this Pull Request

* Adds a new step `difm-design-setup-site` which points to the `design-picker` step component. (`difm` might be renamed to something else in the near future)
* Some changes to the existing design picker:
    * Exclude the blank canvas design if the `excludeBlankCanvas` prop is passed to the step. Context here - pdh1Xd-5k-p2#comment-204
    * Hide the external link in the preview if the `hideExternalPreview` prop is passed. Since this step is before the domains step, we do not yet have a `siteSlug` and we cannot show the URL in the design preview.

### Testing instructions

#### Testing the DIFM Signup Flow.
**Testing as a new user**
* Start the flow at `/start/do-it-for-me?flags=signup/design-picker-categories` in an incognito window.
* After the user step, confirm that the categorized design picker is shown.
    * **Please note that the preview won't work if you are sandboxed. Switch to prod servers to view the preview**.
    * Confirm that the Blank Canvas design is hidden.
    * Confirm that clicking on "Preview <design name>" shows the preview and clicking on "Back" takes you back to the previous screen. Try to play around with going to different previews and clicking on "Back" to ensure that this functionality works well.
    * Confirm that in the Preview screen, the URL is hidden. 
![image](https://user-images.githubusercontent.com/5436027/138687771-51c813ae-6b45-47b7-89fe-bcdd968842b7.png)
* After clicking on "Start with <design>", it should take you to the site info collection step. The form will not load if you are on local calypso. 
   * If testing locally, using React developer tools, check the props passed to the form component. Confirm that the username entered in the first step is passed as a prop correctly.
 
https://user-images.githubusercontent.com/5436027/138688316-cbdf7364-df42-4f8a-a300-2fa1767f8076.mp4
* Click on the "Next" link to skip the form. (This is temporary and will be removed before merging).
* Continue through the flow by selecting a domain.
* After checkout, confirm that the new site has been headstarted with the correct design.

-----

**Testing as an existing user**
* Log in as an existing user.
* Start the flow at `/start/do-it-for-me?flags=signup/design-picker-categories`
* Repeat the testing instructions for a new user. 
* Confirm that the username of the current user is correctly passed as a prop to the form component.

#### Testing for regressions in the `setup-site` flow.
* Go to `/start/setup-site?flags=signup/hero-flow&siteSlug=<your_site>` or `/start?flags=signup/hero-flow`
* In the design picker step, confirm that the blank canvas design is shown.
* Confirm that clicking on "Preview <design" and then clicking on "Back" works correctly.
